#!/usr/bin/env python
PACKAGE = "o2as_phoxi_camera"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# gen.add("size_width", int_t, 1 << 1, "An Integer parameter", 1920, 0)
# gen.add("size_height", int_t, 1 << 2, "An Integer parameter", 1080, 0)
resolution_enum = gen.enum(
    [gen.const("HighResolution", int_t, 0, "A small constant"),
     gen.const("LowResolution",  int_t, 1, "A medium constant")],
    "An enum to set size")
gen.add("resolution",            int_t, 1 << 4,
        "A size parameter which is edited via an enum",
        0, 0, 1, edit_method=resolution_enum)

gen.add("scan_multiplier",    int_t, 1 << 5, "An Integer parameter", 1, 1, 50)
gen.add("shutter_multiplier", int_t, 1 << 6, "An Integer parameter", 1, 1, 20)

trigger_enum = gen.enum(
    [gen.const("Freerun",  int_t, 0, "A small constant"),
     gen.const("Software", int_t, 1, "A medium constant")
#     gen.const("Hardware", int_t, 2, "A large constant"),
#     gen.const("NoValue",  int_t, 3, "An extra large constant")
    ],
    "An enum to set size")
gen.add("trigger_mode",    int_t, 1 << 7,
        "A size parameter which is edited via an enum",
        1, 0, 3, edit_method=trigger_enum)

timeout_enum = gen.enum(
    [gen.const("ZeroTimeout", int_t,  0, "A small constant"),
     gen.const("Infinity",    int_t, -1, "A medium constant"),
     gen.const("LastStored",  int_t, -2, "A large constant"),
     gen.const("Default",     int_t, -3, "An extra large constant")],
    "An enum to set size")
gen.add("timeout",            int_t, 1 << 8,
        "A size parameter which is edited via an enum",
        -1, -3, 0, edit_method=timeout_enum)

gen.add("confidence", double_t, 1 << 9, "A double parameter", 3.0, 0.0, 100.0)

gen.add("send_point_cloud",    bool_t, 1 << 10, "A Boolean parameter", True)
gen.add("send_normal_map",     bool_t, 1 << 11, "A Boolean parameter", True)
gen.add("send_depth_map",      bool_t, 1 << 12, "A Boolean parameter", True)
gen.add("send_confidence_map", bool_t, 1 << 13, "A Boolean parameter", True)
gen.add("send_texture",        bool_t, 1 << 14, "A Boolean parameter", True)

point_format_enum = gen.enum(
    [gen.const("None",  int_t, 0, "A small constant"),
     gen.const("RGB",   int_t, 1, "A medium constant"),
     gen.const("Float", int_t, 2, "A large constant")],
    "An enum to set size")
gen.add("point_format", int_t, 1 << 15,
        "A size parameter which is edited via an enum",
        0, 0, 2, edit_method=point_format_enum)


gen.add("intensity_scale", double_t, 1 << 16, "A double parameter",
        1.0, 0.05, 5.0)

exit(gen.generate(PACKAGE, "o2as_phoxi_camera_node", "o2as_phoxi_camera"))
